ios/Example/Tests/TrifleTests.swift

```diff
@@ -75,34 +96,125 @@ MIIBZTCCAQqgAwIBAgIBATAKBggqhkjOPQQDAjAYMRYwFAYDVQQDDA1pc3N1aW5nRW50aXR5MB4XDTIz
+        let rootCertificate = Certificate(version: 0, certificate: rootCertEncoded)
+
+        // cert chain of length 2
+        let signData1 = try trifle.createSignedData(data: data,
+                                    keyHandle: keyHandle,
+                                    certificate: [deviceCertificate] + [rootCertificate])
+        XCTAssertNotNil(signData1)
+        
+        // cert chain of length 1
+        let signData2 = try trifle.createSignedData(data: data,
+                                     keyHandle: keyHandle,
+                                     certificate: [deviceCertificate] )
+        XCTAssertNotNil(signData2)
+    }
+    
+    func testSignMultipleKeys_succeeds() throws {
+        // NOTE: Right now this test passes because we are not validating
+        // that the certificate contains the public key matching
+        // the signing key
+        // This is TODO - once this is validated, this test should FAIL
+        
+        let data = "hello world".data(using: .utf8)!
+
+        let trifle = try Trifle(reverseDomain: reverseDomain)
+        let keyHandle1 = try trifle.generateKeyHandle()
+        let keyHandle2 = try trifle.generateKeyHandle()
+
+        let deviceCertificate = Certificate(version: 0, certificate: deviceCertEncoded)
+
+        // Key 1
+        let signData1 = try trifle.createSignedData(data: data,
+                                    keyHandle: keyHandle1,
+                                    certificate: [deviceCertificate] )
+        XCTAssertNotNil(signData1)
+        
+        // Key 2
+        let signData2 = try trifle.createSignedData(data: data,
+                                     keyHandle: keyHandle2,
+                                     certificate: [deviceCertificate] )
+        XCTAssertNotNil(signData2)
+    }
+    
+    func testSignMultipleKeysBadDomain_fail() throws {
+        // CHECK BEHAVIOR
+        let data = "hello world".data(using: .utf8)!
+
+        let trifle1 = try Trifle(reverseDomain: reverseDomain)
+        let keyHandle1 = try trifle1.generateKeyHandle()
+
+        let trifle2 = try Trifle(reverseDomain: "app.square.trifle.keys")
+        let keyHandle2 = try trifle2.generateKeyHandle()
+        print("** KEY 1 **\n" + keyHandle1.tag)
+        print("** KEY 2 **\n" + keyHandle2.tag)
+
+        let deviceCertificate = Certificate(version: 0, certificate: deviceCertEncoded)
+
+        // Key 1
+        let signData1 = try trifle1.createSignedData(data: data,
+                                    keyHandle: keyHandle1,
+                                    certificate: [deviceCertificate] )
+        XCTAssertNotNil(signData1)
+        
+        // Key 2
+        let signData2 = try trifle1.createSignedData(data: data,
+                                     keyHandle: keyHandle2,
+                                     certificate: [deviceCertificate] )
+        XCTAssertNotNil(signData2)
+    }
+    
+    
+    func testSignInvalidInput_fail() throws {
```

Comment:
lets reflect what invalid input is. if necessary, split out the unit test

Reply:
oh you mean have the bad root cert as another test. Sounds good.